tests/testsuite/profile_trim_paths.rs

```diff
@@ -511,3 +511,39 @@ fn object_works() {
     assert!(memchr::memmem::find(&stdout, registry_src_bytes).is_none());
     assert!(memchr::memmem::find(&stdout, pkg_root).is_none());
 }
+
+// TODO: might want to move to test/testsuite/build_script.rs once stabilized.
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn custom_build_env_var_trim_paths() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [profile.dev]
+                trim-paths = ["diagnostics", "macro", "object"]
+           "#,
+        )
+        .file("src/lib.rs", "")
+        .file(
+            "build.rs",
+            r#"
+            fn main() {{
+                assert_eq!(
+                    std::env::var("CARGO_TRIM_PATHS").unwrap().as_str(), 
+                    "diagnostics,macro,object"
```

Comment:
We might want to cover `all` and `none` as well.  I'd be tempted to also cover `true`, `false`, and `[]` since we discussed multiple ways of those being implemented which has ramifications for how they get sent to build scripts

Maybe we throw this in a loop?

Reply:
Just added those cases. Thanks for the suggestion!